tests/platform_tests/api/test_fan.py

```diff
@@ -0,0 +1,208 @@
+
+        self.assert_expectations()
+
+    def test_get_fans_speed_tolerance(self, duthost, localhost, platform_api_conn):
+        try:
+            num_fans = int(chassis.get_num_fans(platform_api_conn))
+        except:
+            pytest.fail("num_fans is not an integer")
+
+        fan_list = chassis.get_all_fans(platform_api_conn)
+        pytest_assert(fan_list is not None, "Failed to retrieve fans")
+
+        for i in range(num_fans):
+            fan_instance = chassis.get_fan(platform_api_conn, i)
+            self.expect(fan_instance and fan_instance == fan_list[i], "Fan {} is incorrect".format(i))
+            speed_tol = fan.get_speed_tolerance(platform_api_conn, i)
+            if self.expect(speed_tol is not None, "Unable to retrieve Fan {} speed tolerance".format(i)):
+                if self.expect(isinstance(speed_tol, int), "Fan {} speed tolerance appears incorrect".format(i)):
+                    self.expect(speed_tol > 0 and speed_tol <= 100, "Fan {} speed tolerance {} reading does not make sense".format(i, speed_tol))
+
+        self.assert_expectations()
+
+    def test_set_fans_speed(self, duthost, localhost, platform_api_conn):
+        try:
+            num_fans = int(chassis.get_num_fans(platform_api_conn))
+        except:
+            pytest.fail("num_fans is not an integer")
+
+        fan_list = chassis.get_all_fans(platform_api_conn)
+        pytest_assert(fan_list is not None, "Failed to retrieve fans")
+        target_speed = random.randint(1,100)
+
+        for i in range(num_fans):
+            fan_instance = chassis.get_fan(platform_api_conn, i)
+            self.expect(fan_instance and fan_instance == fan_list[i], "Fan {} is incorrect".format(i))
+            speed = fan.get_speed(platform_api_conn, i)
+
+            speed_tol = fan.get_speed_tolerance(platform_api_conn, i)
+            
+
+            led_status = fan.get_status_led(platform_api_conn, i)
+            speed_set= fan.set_speed(platform_api_conn, i, target_speed)
+            time.sleep(5)
+
+            act_speed = fan.get_speed(platform_api_conn, i)
+            self.expect(abs(act_speed - target_speed)  <= speed_tol, "Fan {} speed change from {} to {} is not within tolerance, actual speed {}".format(i, speed, target_speed, act_speed))
+
+        self.assert_expectations()
+
+    def test_set_fans_led(self, duthost, localhost, platform_api_conn):
+        LED_COLOR_LIST = [
+            "off",
+            "red",
+            "amber",
+            "green",
+        ]
+
+        try:
+            num_fans = int(chassis.get_num_fans(platform_api_conn))
+        except:
+            pytest.fail("num_fans is not an integer")
+
+        fan_list = chassis.get_all_fans(platform_api_conn)
+        pytest_assert(fan_list is not None, "Failed to retrieve fans")
```

Comment:
As above, getting the fan list and comparing against the result of `get_fan(index)` is not necessary here, as this consistency check is already covered by the `test_fans()` function in test_chassis.py.

Reply:
fixed